src/nemo_evaluator/observability/failure_classification.py

```
@@ -0,0 +1,73 @@
+
+
+def _status_from_text(text: str) -> int | None:
+    match = _STATUS_RE.search(text) or _LEADING_STATUS_RE.search(text)
+    return _status_int(match.group("code")) if match else None
+
+
+def _has_code(text: str, *codes: int) -> bool:
+    return re.search(r"\b(" + "|".join(str(code) for code in codes) + r")\b", text) is not None
+
+
+def classify_model_failure(error: str = "", *, status_code: Any = None) -> str | None:
```

Ruff (0.15.20)

[warning] Too many return statements (8 > 6)

(PLR0911)